api/src/main/java/org/bahmni/module/hip/api/dao/HipVisitDao.java

```diff
@@ -0,0 +1,11 @@
+package org.bahmni.module.hip.api.dao;
+
+import java.util.Date;
+import java.util.List;
+
+public interface HipVisitDao {
```

Comment:
Instead of HipVisitDao, can we name it as VisitDao or VisitHelper?

Reply:
I think LabResultsDao goes well as we already have PrecriptionDao and DiagnosticReportDao? Since its returning Ids for both visits and programs "HipVisitDao" seems very generic? and do we need Hip in the name?

Reply:
The idea was to have EncounterDao list out all the encounter ids, and the same encounter list method will be used to fetch data based on the HI Type. The SQL queries have diverged quite a bit so having a generic class name does not mean much now. But if we get to refactor the SQLs eventually, this is the idea structure
DAO
       - VisitDao
       - EncounterDao
       - DiagReportDAO -> calls visit dao to get the ids, fetch diag report for those visit ids
       - LabResultsDao -> calls visit dao to get the visit ids, returns lab results. but we are using an existing Bahmni Core API that converts observation records to lab results which takes visit ids as input. As the returned values from the SQL are visit ids and not lab results, I named the DAO as visitDAO. The problem is while loading hte OMOD due to conflict with another class with same name defined in BahmniCore, I renamed it to HipVISITDAO
       - PrescriptionDao -> calls encounter and gets med data for those encounter ids

Reply:
I will merge the PR for now and will address the class naming convention when refactoring the SQL commands